### PR TITLE
[Misc][Platform] Move use allgather to platform

### PR DESCRIPTION
**Pull request by MengqingCao**
**Mainly changes**

1. Move `use_all_gather` to platform so that the oot devices could override it instead of modifying in vLLM
2. Add check of `sampling_metadata.seq_groups` before applying logits processor, as when using all-gather, the logits is not none, but `sampling_metadata.seq_groups` is none, so `_apply_logits_processors` will raise an unexpected error. We add this check to avoid step into `_apply_logits_processors` when using all-gather

### CODE REVIEW

**Comment by MengqingCao** [comment]
@youkaichao could you help review this? thx!

**Review by youkaichao — APPROVED** [review]
sounds good

**Comment by MengqingCao** [comment]
CI failed due to network issue
